service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java

```diff
@@ -0,0 +1,155 @@
+        for (int i = 0; i < kapuaLocatorErrorCodes.length; i++) {
+            KapuaLocatorException kapuaLocatorException = new KapuaLocatorException(kapuaLocatorErrorCodes[i], argument1, argument2, argument3);
+            assertEquals("Expected and actual values should be the same.", kapuaLocatorErrorCodes[i], kapuaLocatorException.getCode());
+            assertNull("Null expected.", kapuaLocatorException.getCause());
+            assertEquals("Expected and actual values should be the same.", expectedMessage[i], kapuaLocatorException.getMessage());
+            assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaLocatorException.getKapuaErrorMessagesBundle());
+        }
+    }
+
+    @Test
+    public void kapuaLocatorExceptionKapuaLocatorErrorCodesNullObjectParametersTest() {
+        for (int i = 0; i < kapuaLocatorErrorCodes.length; i++) {
+            KapuaLocatorException kapuaLocatorException = new KapuaLocatorException(kapuaLocatorErrorCodes[i], null);
+            assertEquals("Expected and actual values should be the same.", kapuaLocatorErrorCodes[i], kapuaLocatorException.getCode());
+            assertNull("Null expected.", kapuaLocatorException.getCause());
+            assertEquals("Expected and actual values should be the same.", expectedMessageNullArguments[i], kapuaLocatorException.getMessage());
+            assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaLocatorException.getKapuaErrorMessagesBundle());
+        }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void kapuaLocatorExceptionNullKapuaLocatorErrorCodesObjectParametersTest() {
+        KapuaLocatorException kapuaLocatorException = new KapuaLocatorException(null, argument1, argument2, argument3);
+        assertNull("Null expected.", kapuaLocatorException.getCode());
+        assertNull("Null expected.", kapuaLocatorException.getCause());
+        assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaLocatorException.getKapuaErrorMessagesBundle());
+        kapuaLocatorException.getMessage();
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void kapuaLocatorExceptionNullKapuaLocatorErrorCodesNullObjectParametersTest() {
+        KapuaLocatorException kapuaLocatorException = new KapuaLocatorException(null, null);
+        assertNull("Null expected.", kapuaLocatorException.getCode());
+        assertNull("Null expected.", kapuaLocatorException.getCause());
+        assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaLocatorException.getKapuaErrorMessagesBundle());
+        kapuaLocatorException.getMessage();
+    }
+
+    @Test
+    public void kapuaLocatorExceptionKapuaLocatorErrorCodesThrowableObjectParametersTest() {
+        for (Throwable throwable : throwables) {
+            for (int i = 0; i < kapuaLocatorErrorCodes.length; i++) {
```

Comment:
Please use enhanced for loop.

Reply:
I used index i to get values from different arrays in the same position